checker/src/main/java/alluxio/checker/SparkIntegrationChecker.java

```diff
@@ -0,0 +1,293 @@
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import scala.Tuple2;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * A spark job to test the integration of Spark with Alluxio. This class will be triggered
+ * in yarn-client and yarn-cluster mode through spark-submit.
+ *
+ * This job requires a running Spark cluster, but not requires a running Alluxio cluster.
+ * It will check whether Alluxio class and Alluxio filesystem can be recognize in Spark
+ * driver and executors.
+ */
+public class SparkIntegrationChecker {
+  private static final Logger LOG = LoggerFactory.getLogger(SparkIntegrationChecker.class);
+  private static List<Tuple2<Integer, String>> sSparkJobResult;
+  private static boolean sAlluxioHAMode = false;
+  private static String sZookeeperAddress = "";
+
+  /**
+   * Implements check integration of Spark with Alluxio.
+   *
+   * @param sc current JavaSparkContext
+   * @return 0 succeed, 1 fail to recognize Alluxio classes,
+   * 2 fail to recognize Alluxio filesystem, 3 fail to support Alluxio HA mode
+   */
+  private static int checkIntegration(JavaSparkContext sc) {
+    // check if Spark driver can recognize Alluxio classes
+    try {
+      Class alluxioURIClass = Class.forName("alluxio.AlluxioURI");
+      LOG.info("Spark driver can find: " + alluxioURIClass.getName());
+    } catch (ClassNotFoundException e) {
+      LOG.error("Spark driver cannot find AlluxioURI class:" + e);
+      return 1;
+    }
+
+    try {
+      Class alluxioBaseFSClass = Class.forName("alluxio.client.file.BaseFileSystem");
+      LOG.info("Spark driver can find: " + alluxioBaseFSClass.getName());
+    } catch (ClassNotFoundException e) {
+      LOG.error("Spark driver cannot find alluxio.client.file.BaseFileSystem class:" + e);
+      return 1;
+    }
+
+    try {
+      Class alluxioHadoopFSClass = Class.forName("alluxio.hadoop.AlluxioFileSystem");
+      LOG.info("Spark driver can find: " + alluxioHadoopFSClass.getName());
+    } catch (ClassNotFoundException e) {
+      LOG.error("Spark driver cannot find alluxio.hadoop.AlluxioFileSystem class:" + e);
+      return 1;
+    }
+
+    // check if Spark driver can recognize Alluxio filesystem
+    try {
+      Class alluxioFSClass = FileSystem.getFileSystemClass("alluxio", new Configuration());
+      LOG.info("Spark driver can recognize Alluxio filesystem: " + alluxioFSClass.getName());
+    } catch (Exception e) {
+      LOG.error("Spark driver cannot recognize Alluxio filesystem: " + e);
+      return 2;
+    }
+
+    // support Alluxio high availability mode
+    if (alluxio.Configuration.getBoolean(PropertyKey.ZOOKEEPER_ENABLED)) {
+      sAlluxioHAMode = true;
```

Comment:
I don't think you need to cache this value, as it's a single line to get it from Configuration